src/v2/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx

```diff
@@ -0,0 +1,112 @@
+import { Box, Button, Spacer } from "@artsy/palette"
+import React, { FC, Fragment } from "react"
+import {
+  createPaginationContainer,
+  graphql,
+  RelayPaginationProp,
+} from "react-relay"
+import ArtworkGridItemFragmentContainer from "v2/Components/Artwork/GridItem"
+import { Masonry } from "v2/Components/Masonry"
+import { extractNodes } from "v2/Utils/extractNodes"
+import { NewForYouArtworksGrid_viewer } from "v2/__generated__/NewForYouArtworksGrid_viewer.graphql"
+
+interface NewForYouArtworksGridProps {
+  viewer: NewForYouArtworksGrid_viewer
+  relay: RelayPaginationProp
+}
+
+export const NewForYouArtworksGrid: FC<NewForYouArtworksGridProps> = ({
+  viewer,
+  relay,
+}) => {
+  const [loading, setLoading] = React.useState(false)
+  const artworks = extractNodes(viewer.artworksForUser)
+
+  const loadMore = () => {
+    if (!relay.hasMore() || relay.isLoading()) return
+
+    setLoading(true)
+
+    relay.loadMore(25, err => {
+      if (err) console.error(err)
+
+      setTimeout(() => {
+        setLoading(false)
+      }, 2500)
```

Comment:
Why the arbitrary delay?

Reply:
Not sure why I added that, but its probably fine to remove here!